src/importexport/musicxml/internal/musicxml/exportxml.cpp

```diff
@@ -6862,18 +6876,59 @@ void ExportMusicXml::keysigTimesig(const Measure* m, const Part* p)
         }
     }
 
-    TimeSig* tsig = 0;
+    // search all staves for non-generated time signatures
+    std::map<staff_idx_t, TimeSig*> timesigs;   // map staff to time signature
     for (Segment* seg = m->first(); seg; seg = seg->next()) {
         if (seg->tick() > m->tick()) {
             break;
         }
-        EngravingItem* el = seg->element(strack);
-        if (el && el->type() == ElementType::TIMESIG) {
-            tsig = (TimeSig*)el;
+        for (track_idx_t t = strack; t < etrack; t += VOICES) {
+            EngravingItem* el = seg->element(t);
+            if (!el) {
+                continue;
+            }
+            if (el->type() == ElementType::TIMESIG) {
+                LOGN(" found timesig %p tick %d track %zu", el, el->tick().ticks(), el->track());
+                staff_idx_t st = (t - strack) / VOICES;
+                if (!el->generated()) {
+                    timesigs[st] = toTimeSig(el);
+                }
+            }
         }
     }
-    if (tsig) {
-        timesig(tsig);
+
+    // write the time signatues
+    if (!timesigs.empty()) {
+        // determine if all staves have a timesig and all timesigs are identical
+        // in that case a single <time> is written, without number=... attribute
+        size_t nstaves = p->nstaves();
+        bool singleTime = true;
+        // check if all staves have a keysig
+        for (staff_idx_t i = 0; i < nstaves; i++) {
+            if (!muse::contains(timesigs, i)) {
+                singleTime = false;
+            }
+        }
+        // check if all timesigs are identical
+        if (singleTime) {
+            for (staff_idx_t i = 1; i < nstaves; i++) {
+                if (!(timesigs.at(i)->sig() == timesigs.at(0)->sig())) {
+                    singleTime = false;
```

Comment:
I think you could call `break;` here

Reply:
Yes, you could. It wouldn't matter much, as that would only affect parts with more than two staves.